### PR TITLE
Add isEqual() for product-specific price rules

### PR DESCRIPTION
**Pull request by BlackbitNeueMedien**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no

This PR implements the `isEqual()` method for product-specific price rules.

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
@BlackbitNeueMedien tests are failing, can you check?

**Comment by BlackbitNeueMedien** [comment]
Have you ever looked at the tests? Is this failing in https://github.com/coreshop/CoreShop/runs/4869837375?check_suite_focus=true really coming from my change?
> Install Resources for Environment behat.
  0/20 [░                           ]   0% %message%
  4/20 [░░░░░░                      ]  20% Install Class CoreShopCart (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopOrderBundle/CoreShopCart.json)
  6/20 [░░░░░░░░░                   ]  30% Install Class CoreShopOrder (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopOrderBundle/CoreShopOrder.json)
  8/20 [░░░░░░░░░░░░                ]  40% Install Class CoreShopOrderShipment (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopOrderBundle/CoreShopOrderShipment.json)
 10/20 [░░░░░░░░░░░░░░░             ]  50% Install Class CoreShopQuote (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopOrderBundle/CoreShopQuote.json)
 11/20 [░░░░░░░░░░░░░░░░            ]  55% Install Class CoreShopQuoteItem (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopOrderBundle/CoreShopQuoteItem.json)
 12/20 [░░░░░░░░░░░░░░░░░           ]  60% Install Class CoreShopOrderInvoice (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/OrderBundle/Resources/install/pimcore/classes/CoreShopOrderInvoice.json)
 14/20 [░░░░░░░░░░░░░░░░░░░░        ]  70% Install Class CoreShopCustomerGroup (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CustomerBundle/Resources/install/pimcore/classes/CoreShopCustomerGroup.json)
 16/20 [░░░░░░░░░░░░░░░░░░░░░░░     ]  80% Install Class CoreShopCompany (/home/runner/work/CoreShop/CoreShop/src/CoreShop/Bundle/CoreBundle/Resources/install/pimcore/classes/CoreShopCustomerBundle/CoreShopCompany.json)
Error: Process completed with exit code 255.

**Comment by dpfaffenbauer** [comment]
true, can you rebase to latest changes?

**Comment by dpfaffenbauer** [comment]
@BlackbitNeueMedien I think it does fail cause of your changes... it fails trying to install the classes...

**Comment by BlackbitNeueMedien** [comment]
Implemented methods from `TypeDeclarationSupportInterface` which gets extended by `Pimcore\Model\DataObject\ClassDefinition\Data` - however this may have worked before because this PR has not changed anything about this.

**Comment by BlackbitNeueMedien** [comment]
Hmm, still aborting with error code 255. Have no idea... Do you, @dpfaffenbauer ?

**Comment by dpfaffenbauer** [comment]
@BlackbitNeueMedien EqualComparisonInterface was added in a later version of Pimcore 6. CoreShop is tested all the way from Pimcore 6.6 upwards. I just added a BC Layer

File: src/CoreShop/Bundle/ProductBundle/CoreExtension/ProductSpecificPriceRules.php [hunk 1]
```diff
@@ -514,4 +518,24 @@ private function getConfigConditions()
     {
         return $this->getContainer()->getParameter('coreshop.product_specific_price_rule.conditions');
     }
+
+    public function getParameterTypeDeclaration(): ?string
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/ProductBundle/CoreExtension/ProductSpecificPriceRules.php:522]: can you remove this again? we don't support that in 2.2

Comment (BlackbitNeueMedien) [reply src/CoreShop/Bundle/ProductBundle/CoreExtension/ProductSpecificPriceRules.php]: Done, thanks for your support!

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by dpfaffenbauer** [comment]
alright, thanks then :) Maybe you want to add that into the 3.x branch as well? :)

**Comment by BlackbitNeueMedien** [comment]
Oh, I thought all the PRs which get merged to 2.2 will eventually get merged to 3.x?

**Comment by dpfaffenbauer** [comment]
Merging things upwards to 3.x/master is not possible easily. To much has changed between the two versions already.

**Comment by dpfaffenbauer** [comment]
@BlackbitNeueMedien managed to merge it

**Pull request merged**